packages/components/tabs/src/tab-group.ts

```diff
@@ -32,380 +30,351 @@ const OBSERVER_OPTIONS: MutationObserverInit = {
   @property({ type: Boolean, reflect: true }) vertical?: boolean;
 
-  /** The alignment of tabs inside sl-tab-group  */
-  @property({ reflect: true }) alignment: TabsAlignment = 'start';
+  override connectedCallback(): void {
+    super.connectedCallback();
 
-  /** The listbox element with all tabs list. */
-  @query('[popover]') listbox!: HTMLElement;
+    this.#mutationObserver.observe(this, OBSERVER_OPTIONS);
 
-  /** Get the selected tab button, or the first tab button. */
-  get #initialSelectedTab(): Tab | null {
-    return this.querySelector('sl-tab[selected]') || this.querySelector('sl-tab:not([disabled])');
+    // We need to wait for the next frame so the element has time to render
+    requestAnimationFrame(() => {
+      const tablist = this.renderRoot.querySelector('[part="tablist"]') as Element;
+
+      // We want to observe the size of the tablist, not the
+      // container or wrapper. The tablist is the element that
+      // changes size for example when fonts are loaded. The
+      // other elements do not change size while the tablist does.
+      this.#resizeObserver.observe(tablist);
+    });
   }
 
-  override render(): TemplateResult {
-    this.#moreButton = this.renderRoot.querySelector('#more-btn') as HTMLButtonElement;
-    this.#allTabs = Array.from(this.querySelectorAll('sl-tab'))?.map(tab => tab.cloneNode(true)) as Tab[];
-    this.#allTabs.forEach(tab => tab.classList.add('listbox-tab'));
+  override disconnectedCallback(): void {
+    this.#resizeObserver.disconnect();
+    this.#mutationObserver.disconnect();
 
+    super.disconnectedCallback();
+  }
+
+  override updated(changes: PropertyValues<this>): void {
+    super.updated(changes);
+
+    if (changes.has('alignTabs')) {
+      this.#shouldAnimate = false;
+      this.#updateSelectionIndicator();
+      this.#shouldAnimate = true;
+    }
+  }
+
+  override render(): TemplateResult {
     return html`
-      <div class="container" part="container" @keydown=${this.#handleKeydown}>
-        <div class="wrapper" part="wrapper">
-          <div @click=${this.#handleTabChange} role="tablist" part="tab-list">
-            <span class="indicator" role="presentation"></span>
-            <slot name="tabs" @slotchange=${() => this.#rovingTabindexController.clearElementCache()}></slot>
-            <div
-              ${anchor({ element: this.#moreButton, position: this.vertical ? 'bottom-start' : 'bottom-end' })}
-              @click=${this.#handleTabChange}
-              @toggle=${this.#onToggle}
-              aria-labelledby="more-btn"
-              id="tabs-popover"
-              part="listbox"
-              popover
-              role="listbox"
-            >
-              ${this.#allTabs}
+      <div part="container">
+        <div part="wrapper">
+          <div class="fade-container">
+            <div class="fade fade-left"></div>
+            <div class="fade fade-right"></div>
```

Comment:
can't this be done with pure css? `::before` and `::after`?

Reply:
Unfortunately not. The reason for this is also why both `.fade-left` and `.fade-right` are *before* the scroller element: if you hover over a tab, that tab changes `z-index` so it's above the fade elements. If you use `::after`, then you can't get above the fade (same as placing `.fade-right` after the scroller).